src/test/java/org/example/automation/amazon/VerifyFilterSearchTest.java

```diff
@@ -0,0 +1,137 @@
+package org.example.automation.amazon;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindAll;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VerifyFilterSearchTest extends BaseWebDriverTest{
+    public static final String WEB_ADDRESS="https://www.amazon.com";
+    public static final String BRAND_APPLE="Apple";
+    public static final String BRAND_ROKU="Roku";
+
+    @FindBy(xpath = "//a[@aria-label='Computers & Accessories']")
+    private WebElement filterCategory;
+
+    @FindBy(xpath = "//div[@class='a-section a-spacing-none']/descendant::span[.='"+BRAND_APPLE+"']")
+    private WebElement filterBrand;
+
+    @FindBy(xpath = "//div[@class='a-section a-spacing-none']/descendant::span[.='"+BRAND_ROKU+"']")
+    private WebElement filterBrandRoku;
+    @FindBy(xpath = "//a/span[.='$25 to $50']")
+    private WebElement filterPrice;
+    @FindAll({@FindBy(xpath = "//div[@class='s-main-slot s-result-list s-search-results sg-row']/descendant::span[@class='a-size-base-plus a-color-base a-text-normal']")})
+    public List<WebElement> productTitleList;
+
+    @FindAll({@FindBy(xpath = "//div[@class='s-main-slot s-result-list s-search-results sg-row']/descendant::a/span[@class='a-price']/span[@class='a-offscreen']")})
+    public List<WebElement> productPriceList;
+
+    @FindBy(xpath = "//span[@aria-label='Sort by:']/descendant::span[.='Featured']")
+    public WebElement sortFeatured;
+
+    @FindBy(xpath = "//li[@aria-labelledby='s-result-sort-select_2']/a[.='Price: High to Low']")
+    public WebElement sortHighToLow;
+
+    @BeforeMethod
+    public void setup() {
+        PageFactory.initElements(webDriver, this);
+        webDriver.get("https://www.amazon.com");
+
+        filterCategory.click();
+    }
+
+    @Test (description = "test case to check if products shown has Apple word inside to " +
+            "prove that the brand category filter working properly")
+    public void verifyBrandCategory() {
+        filterBrand.click();
+        int check=0;
+        for(WebElement e : productTitleList) {
+            System.out.println(e.getText());
+            if (!e.getText().contains(BRAND_APPLE)){
+               check++;
+            }
+        }
+        Assert.assertEquals( check,0);
+    }
+
+    @Test (description = "test case to check if multiple brand category filter working properly")
+    public void verifyMultipleBrandCategory() {
+        filterBrand.click();
+        filterBrandRoku.click();
+        boolean check= true;
+        for(WebElement e : productTitleList) {
+            System.out.println(e.getText());
+            if (e.getText().contains(BRAND_APPLE)||e.getText().contains(BRAND_ROKU)){
```

Comment:
I suggest following code:

 if (!e.getText().contains(BRAND_APPLE)||!e.getText().contains(BRAND_ROKU)){
                check=false;
                break;
            }

Reply:
It will not work because that means products that is not Apple will directly cause check=false. While it can be because the product has Roku brand.
Example:
Roku Keyboard=> !e.getText().contains(BRAND_APPLE) ( this will be correct since product brand is not Apple)=> Ans since or only need 1 of them correct thus check=false (eventhought it should not be false since it has ROKU brand)